specify coinbase validity rules (#442)
FuelLabs/fuel-specs: src/protocol/tx_validity.md

### State Changes

Transaction processing is completed by removing spent UTXOs from the state and adding created UTXOs to the state.

### Coinbase

The coinbase is a mechanism for block creators to convert fees into spendable UTXOs.

In order for a coinbase to be valid:

> 1. It must issued in the form of a [Mint](../protocol/tx_format/transaction.md#TransactionMint) transaction.

Reviewer: ```suggestion
1. It must be a [Mint](../protocol/tx_format/transaction.md#TransactionMint) transaction.
```

> 2. The coinbase must be the first transaction within a block.

Reviewer 2: ```suggestion
2. The coinbase must be the first transaction within a block, even if there are no other transactions in the block and the fee is zero.
```

Reviewer 2: It is how it works now. We can make it optional, but it will require patching the existing logic, and we need to keep it in mind during work with block. 

If the most often case for us is non-zero coinbase transactions, then making it optional may not be worth it because the more complex validation rules could cause bugs in the code.

Reviewer: Alright. Both approaches are viable, so whatever works better for you guys.

3. The total output value of the coinbase cannot exceed the total amount of fees processed from all other transactions within the same block.

> 4. The `asset_id` for coinbase outputs must match the `asset_id` that fees are paid in (`asset_id == 0`).

Reviewer 2: ```suggestion
4. The `asset_id` for coinbase outputs must match the `asset_id` that fees are paid in (`asset_id == 0` by default).
```

Reviewer: I wouldn't say this is a default; rather, it's a requirement that the asset ID of the coinbase tx output MUST be something. That something could change later, but it's still a requirement.

Reviewer 2: I meant that the default asset to pay fee is `0`=) But it can be different for different networks in the future(I don't know why, but who knows)

Author: closing this thread as we use asset_id = 0 widely throughout the specs to refer to the base asset. If we want to make this more configurable that would be a larger, separate change.

> 4. The `asset_id` for coinbase transaction outputs must match the `asset_id` that fees are paid in (`asset_id == 0`).

Reviewer: Thoughts: we should have a parameters page, and make the base asset ID a parameter.
